java-checks/src/main/java/org/sonar/java/checks/helpers/ConstantsHelper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.helpers;
+
+import org.sonar.java.model.LiteralUtils;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+public class ConstantsHelper {
+
+  private ConstantsHelper() {
+  }
+
+  public static boolean isStringLiteralWithValue(Tree tree, String expectedValue) {
+    if (tree == null || expectedValue == null || !tree.is(Tree.Kind.STRING_LITERAL)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this expression which always evaluates to "false" ([squid:S2589](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=2009&issues=AWNEwxEiOa7A50ZeJHHl&open=AWNEwxEiOa7A50ZeJHHl)

[ISSUE_KEY]: AWNEwxEiOa7A50ZeJHHl